brave/utility/tor/tor_impl.cc

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2018 The Brave Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "brave/utility/tor/tor_impl.h"
+
+#include <utility>
+
+#include "base/command_line.h"
+#include "base/process/kill.h"
+#include "base/process/launch.h"
+#include "base/single_thread_task_runner.h"
+#include "base/task_scheduler/post_task.h"
+#include "base/threading/thread.h"
+#include "base/threading/thread_task_runner_handle.h"
+
+#if defined(OS_POSIX)
+int pipehack[2];
+
+static void SIGCHLDHandler(int signal) {
```

Comment:
Tiny nit: there is a standard library routine `signal`; some compilers will complain about shadowing names like that (`-Wshadow`).  So I usually use the variable name `signo` here.  Not necessary to change now because obviously this must work as is, but wouldn't hurt for future-proofing compiler and warning choices.

Reply:
fixed in 4e72ec6